Sources/WalletConnectSign/Engine/Common/SessionEngine.swift

```diff
@@ -235,8 +242,21 @@ private extension SessionEngine {
         guard !request.isExpired() else {
             return respondError(payload: payload, reason: .sessionRequestExpired, protocolMethod: protocolMethod)
         }
-
-        onSessionRequest?(request)
+        
+        if let rawRequest = payload.rawRequest, let verifyClient {
+            Task(priority: .high) {
+                let attestationId = rawRequest.rawRepresentation.sha256().toHexString()
+                let origin = try? await verifyClient.registerAssertion(attestationId: attestationId)
+                let sessionContext = await Session.Context(
+                    origin: origin,
+                    validation: (origin == session.peerParticipant.metadata.url) ? .valid : (origin == nil ? .unknown : .invalid),
+                    verifyUrl: verifyClient.verifyHost
```

Comment:
same here. `verifyOrigin` and I would handle it generically through Networking package

Reply:
So I would add `verifiable` property to `ProtocolMethod`, call `verifyOrigin` in NetworkingInteractor and pass VerifyContext to requestSubscription

@flypaper0 what do you think?